diff --git a/arch/arm/boot/dts/imx6ul.dtsi b/arch/arm/boot/dts/imx6ul.dtsi
index 62ed30c..a685694 100644
--- a/arch/arm/boot/dts/imx6ul.dtsi
+++ b/arch/arm/boot/dts/imx6ul.dtsi
@@ -942,6 +942,14 @@
 					reg = <0x38 4>;
 				};
 
+				pxp: pxp@21cc000 {
+					compatible = "fsl,imx6ul-pxp";
+					reg = <0x021cc000 0x4000>;
+					interrupts = <GIC_SPI 8 IRQ_TYPE_LEVEL_HIGH>;
+					clock-names = "axi";
+					clocks = <&clks IMX6UL_CLK_PXP>;
+				};
+
 				tempmon_temp_grade: temp-grade@20 {
 					reg = <0x20 4>;
 				};
diff --git a/arch/arm/boot/dts/imx6ull.dtsi b/arch/arm/boot/dts/imx6ull.dtsi
index f3668fe..70a6bce 100644
--- a/arch/arm/boot/dts/imx6ull.dtsi
+++ b/arch/arm/boot/dts/imx6ull.dtsi
@@ -67,3 +67,9 @@
 		};
 	};
 };
+
+&pxp {
+	compatible = "fsl,imx6ull-pxp";
+	interrupts = <GIC_SPI 8 IRQ_TYPE_LEVEL_HIGH>,
+ 	<GIC_SPI 18 IRQ_TYPE_LEVEL_HIGH>;
+};
\ No newline at end of file
